share/completions/gsettings.fish

```diff
@@ -0,0 +1,99 @@
+# Fish completions for glibs's "gsettings" configuration tool
+
+function __fish_complete_gsettings_args
+    set -l schema_commands get monitor writable range describe set reset reset-recursively list-keys list-children list-recursively
+    set -l key_commands get monitor writable range describe set reset
+
+    set -l schemadir
+    set -l cmd (commandline -poc)
+    set -e cmd[1]
+
+    if set -q cmd[2]; and string match -q -- '--schemadir' $cmd[1]
+        # use eval to ensure the expansion of ~
+        set schemadir --schemadir (eval echo $cmd[2])
+        set -e cmd[1..2]
+    end
+
+    if not set -q cmd[1]
+        return 1
+    end
+
+    set -l subcommand $cmd[1]
+
+    if not contains -- $subcommand $schema_commands
+        return 1
+    end
+
+    if not set -q cmd[2]
+        # Non-relocatable schemas
+        gsettings $schemadir list-schemas ^/dev/null
+
+        # Relocatable schemas require a dconf path, but there is no universal way of
+        # finding the right paths to suggest here. Just default to '/'.
+        gsettings $schemadir list-relocatable-schemas ^/dev/null | while read -l line
+            echo $line:/
+        end
+        return 0
+    end
+
+    set -l schema $cmd[2]
+
+    if not contains -- $subcommand $key_commands
+        return 1
+    end
+
+    if not set -q cmd[3]
+        gsettings $schemadir list-keys $schema ^/dev/null
+        return 0
+    end
+
+    set -l key $cmd[3]
```

Comment:
In fact it should be put inside the `if not set -q cmd[4]` block since that is the only place the var is used.